test/SlowTests/Issues/RavenDB-17650.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Amazon.S3.Model;
+using FastTests;
+using Microsoft.Azure.Documents.Spatial;
+using Raven.Client.Documents.Subscriptions;
+using Raven.Client.ServerWide.Operations;
+using Raven.Server.Config;
+using SlowTests.MailingList;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_17650 : ClusterTestBase
+    {
+        public RavenDB_17650(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task Should_Retry_When_DatabaseDisabledException_Was_Thrown()
+        {
+            using var store = GetDocumentStore(new Options()
+            {
+                ReplicationFactor = 1,
+                RunInMemory = false
+            });
+            
+            string id = "User/33-A";
+            using (var session = store.OpenAsyncSession())
+            {
+                await session.StoreAsync(new User { Id = id, Name = "1" });
+                await session.StoreAsync(new User { Name = "2" });
+                await session.SaveChangesAsync();
+            }
+
+            await store.Subscriptions
+                .CreateAsync(new SubscriptionCreationOptions<User>
+                {
+                    Name = "BackgroundSubscriptionWorker"
+                });
+            
+            using var worker = store.Subscriptions.GetSubscriptionWorker<User>(new SubscriptionWorkerOptions("BackgroundSubscriptionWorker"));
+
+            // disable database
+            var disableSucceeded = store.Maintenance.Server.Send(new ToggleDatabasesStateOperation(store.Database, true));
+            Assert.True(disableSucceeded.Success);
+            Assert.True(disableSucceeded.Disabled);
+
+            var cts = new CancellationTokenSource();
+            var mre = new ManualResetEvent(false);
+            var t = worker.Run(async batch =>
+            {
+                mre.Set();
+            }, cts.Token);
+
+            //enable database
+            await Task.Delay(500);
```

Comment:
why do we need a delay here?

Reply:
Because I want to be sure that the subscription worker tried to approach the server/database and failed before I enable the database/revive the servers (nodes).